server/src/main/java/io/crate/expression/predicate/IsNullPredicate.java

```diff
@@ -122,7 +122,7 @@ public Query toQuery(Function function, Context context) {
             if (refExistsQuery == null) {
                 return null;
             }
-            if (ref.valueType() instanceof ArrayType<?> && refExistsQuery instanceof FieldExistsQuery) {
+            if (ref.valueType() instanceof ArrayType<?>) {
```

Comment:
Initially  I had only `if (objType.innerTypes().isEmpty())` check which was enough to address
 `ARRAY(OBJECT)` with `[]` value and  `OBJECT` with `{}` value. 
 
However, we can have empty arrays/objects in a object actually having some sub-columns, so this change actually fixes all cases and change below is just small optimization.